<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>684</width>
    <height>568</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>压缩文件密码破解器</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="2" column="0">
     <widget class="QGroupBox" name="groupBox_external_dict">
      <property name="title">
       <string>自定义字典</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>请选择自定义字典</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_dict_path"/>
       </item>
       <item>
        <widget class="QToolButton" name="toolButton_dict_path">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_crack">
      <property name="title">
       <string>开始破解</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>请选择压缩文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_file_path"/>
         </item>
         <item>
          <widget class="QToolButton" name="toolButton_file_path">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>请选择解压位置</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_extract_path"/>
         </item>
         <item>
          <widget class="QToolButton" name="toolButton_extract_path">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item alignment="Qt::AlignLeft">
          <widget class="QPushButton" name="pushButton_crack">
           <property name="text">
            <string>开始破解</string>
           </property>
          </widget>
         </item>
         <item alignment="Qt::AlignLeft">
          <widget class="QLabel" name="label_password_is">
           <property name="text">
            <string>密码是</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_password">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QRadioButton" name="radioButton_select_dict">
        <property name="text">
         <string>使用内置字典</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="radioButton_customized_dict">
        <property name="text">
         <string>使用自定义字典</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_internal_dict">
      <property name="title">
       <string>内置字典</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QCheckBox" name="checkBox_num">
             <property name="text">
              <string>数字</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBox_lower_letter">
             <property name="text">
              <string>小写字母</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBox_upper_letter">
             <property name="text">
              <string>大写字母</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBox_symbols">
             <property name="text">
              <string>特殊符号</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item alignment="Qt::AlignRight">
            <widget class="QLineEdit" name="lineEdit_digit2">
             <property name="maximumSize">
              <size>
               <width>59</width>
               <height>16777215</height>
              </size>
             </property>
            </widget>
           </item>
           <item alignment="Qt::AlignLeft">
            <widget class="QLabel" name="label_digit1">
             <property name="text">
              <string>位</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="layoutDirection">
              <enum>Qt::RightToLeft</enum>
             </property>
             <property name="text">
              <string>至</string>
             </property>
            </widget>
           </item>
           <item alignment="Qt::AlignRight">
            <widget class="QLineEdit" name="lineEdit_digit1">
             <property name="maximumSize">
              <size>
               <width>72</width>
               <height>16777215</height>
              </size>
             </property>
            </widget>
           </item>
           <item alignment="Qt::AlignLeft">
            <widget class="QLabel" name="label_digit2">
             <property name="text">
              <string>位</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>请选择字典保存位置</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_save_dict"/>
         </item>
         <item>
          <widget class="QToolButton" name="toolButton_save_dict">
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_write_dict">
           <property name="maximumSize">
            <size>
             <width>167</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>生成字典</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_write_dict</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>button_write_dict_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>567</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_save_dict</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>button_save_dict_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>513</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>563</x>
     <y>135</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_select_dict</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>change_dict()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_customized_dict</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>change_dict()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>576</x>
     <y>312</y>
    </hint>
    <hint type="destinationlabel">
     <x>575</x>
     <y>357</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_dict_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>button_select_dict_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>601</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_file_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_zip_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>632</x>
     <y>406</y>
    </hint>
    <hint type="destinationlabel">
     <x>681</x>
     <y>409</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_extract_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_unzip_dir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>652</x>
     <y>457</y>
    </hint>
    <hint type="destinationlabel">
     <x>681</x>
     <y>457</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_crack</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>button_crack_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>81</x>
     <y>507</y>
    </hint>
    <hint type="destinationlabel">
     <x>76</x>
     <y>541</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>button_write_dict_clicked()</slot>
  <slot>button_save_dict_clicked()</slot>
  <slot>change_dict()</slot>
  <slot>button_select_dict_clicked()</slot>
  <slot>select_zip_file()</slot>
  <slot>select_unzip_dir()</slot>
  <slot>button_crack_clicked()</slot>
 </slots>
</ui>
